src/server/main_service.cc

```diff
@@ -892,6 +895,11 @@ void Service::DispatchCommand(CmdArgList args, facade::ConnectionContext* cntx)
   }
 
   uint64_t start_ns = ProactorBase::GetMonotonicTimeNs(), end_ns;
+  double oom_deny_ratio = GetFlag(FLAGS_oom_deny_ratio);
+  uint64_t used_memory = ServerState::tlocal()->GetUsedMemory(start_ns);
```

Comment:
1. we already have `etl` for `ServerState::tlocal()`.
2. Lets check first if the mask has DENYOOM before fetching all the data. We are on the hotpath.
3. @chakaz you mentioned that GetFlag is heavy. is it indeed the case?